### DISCUSSION TITLE
Using service role secret key with supabase-js on the server

### DISCUSSION THREAD

**Discussion by SandroMaglione** [supabase/supabase#1284] [Questions] [answered]
Hi,

I am working on a Stripe webhook that should insert and update data on a row-level security restricted table.
On the server, the JWT of the client is not available, so I suppose this is the perfect usecase for the service role secret key.

```ts
// No JWT, operation restricted!
await supabase
            .from<definitions['subscription']>('subscription')
            .select()
            .match({
              id: user.id,
              subscription_id: subscriptionId,
            });
```

How can I pass the secret key using supabase-js?

**Comment by inian** [comment, marked as answer]
The service role can be passed as part of the [createClient call](https://supabase.io/docs/reference/javascript/initializing#createclient)

```const supabase = createClient(supabaseURL, serviceKey);```

**Reply by SandroMaglione** [reply]
Great, it works, thank you!

It was not really clear to me from the docs. Do you think it would be appropriate to add a clarification somewhere?

**Reply by inian** [reply]
Yeah, the docs mention anon key everywhere to be safe. If you have suggestions on this, let me know :)

**Reply by fergusmeiklejohn** [reply]
> Yeah, the docs mention anon key everywhere to be safe. If you have suggestions on this, let me know :)

I've just been searching for the same information. I wonder if we should have a Node section in the docs? Or maybe you are already working on a nodejs admin library to go with Functions?

**Reply by sophiabk** [reply]
> > Yeah, the docs mention anon key everywhere to be safe. If you have suggestions on this, let me know :)
> 
> I've just been searching for the same information. I wonder if we should have a Node section in the docs? Or maybe you are already working on a nodejs admin library to go with Functions?

Agree this is unclear. The security around the anon key is clear in the docs but where to actually use it is not.

**Reply by mrChernicharo** [reply]
great answer! Works great in dev!
However I'm having trouble in prod.`Uncaught Error: supabaseKey is required` 

Still struggling to figure out how to interact with auth.users (update/delete stuff there) from actions in the client...
Any helpful resources would be appreciated

**Reply by devidw** [reply]
@mrChernicharo when this is working for you in dev but not in prod, this is probably due to the availability of environment variables? Make sure your service role key is set in your prod environment.

**Comment by pushkarsingh32** [comment]
How to do it in auth helper package unable to see any doc explaining it

**Reply by ghost** [reply]
```js
const cookieStore = cookies()
// Requires special privileges to use admin functions, so we need to use the service_role_key
const {SUPABASE_SERVICE_KEY} = process.env
const supabase = createRouteHandlerClient({ cookies: () => cookieStore }, { supabaseKey: SUPABASE_SERVICE_KEY })
```
